src/main.cpp

```diff
@@ -64,70 +45,61 @@ void loop()
   delay(100);
 
   // check if there is a card
-  if (mfrc522.PICC_IsNewCardPresent())
+  if (board.getRfid().PICC_IsNewCardPresent())
   {
     // if there is a "new" card (could be the same that stayed in the field)
-    if (!mfrc522.PICC_ReadCardSerial() || !ready_for_a_new_card)
+    if (!board.getRfid().PICC_ReadCardSerial() || !ready_for_a_new_card)
     {
       return;
     }
     ready_for_a_new_card = false;
 
     // Acquire the UID of the card
-    candidate_user.setUidFromArray(mfrc522.uid.uidByte);
+    auto uid = board.getRfid().uid.uidByte;
 
-    if (machine.isFree())
+    if (board.getMachine().isFree())
     {
       // machine is free
-      if (server.isAuthorized(candidate_user))
+      if (board.authorize(uid))
       {
-        machine.login(candidate_user);
-        LCD.state(LCDState::LOGGED_IN);
-        LCD.update(server, candidate_user, machine);
-        delay(1000);
-      }
-      else
-      {
-        LCD.state(LCDState::LOGIN_DENIED);
-        LCD.update(server, candidate_user, machine);
-        delay(1000);
+        Serial.println("Login successfull");
+      } else {
+        Serial.println("Login failed");
       }
+      delay(1000);
     }
     else
     {
       // machine is busy
-      if (machine.getActiveUser().getUid() == candidate_user.getUid())
+      FabMember member(uid);
+
+      if (board.getMachine().getActiveUser().getUid() == member.getUid())
       {
         // we can logout. we should require that the card stays in the field for some seconds, to prevent accidental logout. maybe sound a buzzer?
-        machine.logout();
-        LCD.state(LCDState::LCDStateType::LOGOUT);
-        LCD.update(server, candidate_user, machine);
+        board.logout();
       }
       else
       {
         // user is not the same, display who is using it
-        LCD.state(LCDState::ALREADY_IN_USE);
-        LCD.update(server, candidate_user, machine);
+        board.changeStatus(BoardStatus::ALREADY_IN_USE);
         delay(1000);
       }
       delay(1000);
     }
   }
   else
   {
-    Serial.println(mfrc522.PICC_IsNewCardPresent() ? "New card" : "No card");
+    Serial.println(board.getRfid().PICC_IsNewCardPresent() ? "New card" : "No card");
     ready_for_a_new_card = true; // we should get SOME "no card" before flipping this flag
     // print status on lcd screen
 
-    if (!machine.isFree())
+    if (!board.getMachine().isFree())
     {
-      LCD.state(LCDState::IN_USE);
-      LCD.update(server, candidate_user, machine);
+      board.changeStatus(BoardStatus::IN_USE);
```

Comment:
you use board.getMachine().isFree(); to check if board.changeStatus(BoardStatus::IN_USE);
this is strange, as you could change board.changeStatus(BoardStatus::IN_USE); when you also call the function that  changes board.getMachine().isFree(); (aka, board.autorise() and logout() should be the one calling changeStatus();

That should completely get rid of the else{} when you have no card readed